packages/contracts-bedrock/src/cannon/interfaces/IPreimageOracle.sol

```diff
@@ -72,5 +72,5 @@ interface IPreimageOracle {
 
     /// @notice Prepares a point evaluation precompile result to be read by the keccak256 of its input.
     /// @param _input The point evaluation precompile input.
-    function loadKZGPointEvaluationPreimage(bytes calldata _input) external;
+    function loadKZGPointEvaluationPreimagePart(uint256 _partOffset, bytes calldata _input) external;
```

Comment:
The preimage data that is read is always 1 byte, hence I omitted the `_partOffset` in the original draft. Are you sure we need to add this parameter? Can't the challenger just call the function the without offset to prepare the oracle?

Reply:
it's awkward.

Although the precompile result is a 1-byte value, the program/VM reads this result in 3 parts. Twice for each 4-byte preimage length part, and once for the precompile result. This means we have 3 steps, with states containing preimageOffsets of 0, 4 and 8 for the entire preimage read. Hence the need for `_partOffset`.

So the only way to avoid loading each part separately is to have `PreimageOracle.loadKZGPointEvaluationPreimage` do this internally. So something like:
```solidity
preimagePartOk[key][0] = true;
preimagePartOk[key][4] = true;
preimagePartOk[key][8] = true;
```
This way op-challenger only needs a single call to the oracle in order to cover the 3 steps.

I don't like this for 2 reasons:
- it deviates from the way we prepare preimages. While it's efficient (as we avoid calling the precompile thrice if 3 consecutive states are in dispute), it's doesn't seem worth optimizing for as it's an unlikely scenario.
- Technically restricts the space of op-programs. Any state with a `PreimageOffset=1` becomes invalid. A preimage read that's out of alignment is technically allowed by the VM. But hardcoding the offset means the op-program must always perform aligned reads. This isn't a real issue in practice, but it's nice to not to deviate from the VM spec if we can help it.

Reply:
ah I see, then yes, we need the `_partOffset`, so we can do a single `preimagePartOk` write at a time per step